### ISSUE TITLE
Sprout doesn't start using new instances of Astaire until Sprout is restarted

### ISSUE THREAD

**Issue by MatMeredith** [Metaswitch/sprout#1767]
I've been trying to fix clearwater-docker after the addition of Astaire.   Something I've noticed is that if I have a deployment with a single Sprout and a single Astaire, and then bring up a second instance of Astaire, Sprout doesn't seem to start using the second Astaire instance until I restart the Sprout process (and then it starts using it straight away).  

Prior to restarting, pinging Astaire from Sprout is alternately returning the IP addresses of each Astaire instance, so DNS is set up correctly.   Is e.g. Sprout only resolving the hostname at start of day or something?

Running off latest code as of Friday March 3rd.

**Comment by mirw** [comment]
Yes, sprout's memcached client does not use Clearwater's DNS resolver, so I suspect it won't cache hostnames in the same way that other protocols do.  I believe this is a new bug that comes from proxying through Astaire - previously Sprout was configured directly with all its memcached server instances' IP addresses.
